COG Support

jbouffard opened the pull request.

Commit:
Allow catalog readers to use COG backends

Signed-off-by: jpolchlo <jpolchlopek@azavea.com>

Cleaned and fixed the logic for reading/writing COGLayers

Added new tests files

Created the COGTest

Bumped the GeoTrellis version to 2.1.0-SNAPSHOT

Cleaned and fixed the logic in the backend some more

Cleaned up the tests

Updated the docstring to the write function

Removed unused imports

Updated the update method to use COGLayerWriter

Added the -v flag to the test_script

Changed some of the Travis test parameters

Cleaned up the cog_test

Cleaned up tests and marked some to be skipped on Travis

Reverted some of the changes made to the Travis test parameters

jpolchlo commented on geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/io/ValueReaderWrapper.scala:
```diff
@@ -21,43 +21,24 @@ import java.time.ZonedDateTime
 import java.util.ArrayList
 import scala.collection.JavaConverters._
 import scala.collection.mutable
+import scala.util.Try
```
Imported, never used

jbouffard replied:
Thanks for the catching that

jbouffard committed:
Improved the to_geotiff_rdd method so that it produces GeoTiffs with internal overviews

Updated the to_geotiff_rdd methods' docstrings

Updated the to_geotiff_rdd_test

Changed the default value for the decimation parameter so that no internaloverviews are built by defualt

Cleaned up the code in the backend

jbouffard committed:
Merge pull request #691 from jbouffard/improvement/geotiff-rdd

to_geotiff_rdd Will Now Produce GeoTiffs that Can Be Used as COGs

jbouffard committed:
Bumped the GeoTrellis version to 2.1.0

jbouffard committed:
Fixed pyramiding_test
